konveyor/crane
# Add support for ingress and automation for developers to test on minikube

This depends on https://github.com/konveyor/crane-lib/pull/93

In order to test this execute the following steps:
1. install jq, minikube and kvm driver for minikube in the system
2. run `bash hack/minikube-clusters.sh`
3. wait for the ingress pods to be running with `kubectl get pods -n ingress-nginx --context=destination-cluster`
4. Create your workloads and identify 1 PVC to sync
5. Build crane with this PR and run `./crane  transfer-pvc --destination-context=destination-cluster --source-context=source-cluster --pvc-name mysql --pvc-namespace mysql`
6. You will see logs like this if the run is successful
```
2021/12/03 22:00:49 [114] Total bytes sent: 67.32K
Total bytes sent: 67.32K
2021/12/03 22:00:49 [114] Total bytes received: 211.77K
Total bytes received: 211.77K

2021/12/03 22:00:49 [114] sent 67.32K bytes  received 211.77K bytes  558.17K bytes/sec
sent 67.32K bytes  received 211.77K bytes  558.17K bytes/sec
2021/12/03 22:00:49 [114] total size is 8.60K  speedup is 0.03
total size is 8.60K  speedup is 0.03
```
7. clean up the environment by running `bash hack/delete-cluster.sh`
